### ISSUE TITLE
"no-redeclare" does not work in node 0.10 with ES6 code

### ISSUE THREAD

**Issue by koorchik** [eslint/eslint#4490]
"babel-eslint": "^4.1.5"
"eslint": "^1.9.0"

Code to verify
```javascript
const {type, ...otherParams} = {};
```

.eslintrc
```javascript
{
    "parser": "babel-eslint",

    "ecmaFeatures": {
        "modules": true
    },

    "rules":{
        "no-redeclare": 2,
    }
}
```

fails with errors
```
  1:17  error  "otherParams" is already defined  no-redeclare
  1:17  error  "otherParams" is already defined  no-redeclare

✖ 2 problems (2 errors, 0 warnings)

```

The latest versions of nodejs (node 5.1, for example) will not throw the error

**Comment by nzakas** [comment]
Duplicate of #4481
